bazel/external/requirements.in

```diff
@@ -30,11 +30,22 @@ h2>=3,<4
 #  priority==1.3.0 is requirement of aioh2 (loaded via bazel)
 priority==1.3.0
 requests
-pycryptodome
+pycryptodome>=3.9.9
 spyne
 aiohttp
 lxml==4.7.1
 typing_extensions
+eventlet==0.30.2
```

Comment:
I think the idea of `requirements.in` is to not pin the version numbers (only add `>=` and `<` contraints if necessary) and let the versions be pinned only in the `requirements.txt` generated by `pip-compile` as described here: https://github.com/magma/magma/blob/6385bd05f3e81b04492ec6065f0d19597b0d4ab8/bazel/external/requirements_README.md?plain=1#L5

Reply:
Why should versions not be pinned via "=="? How would the compile step know if the system depends on a fixed version?

Reply:
If we pin dependencies in `requirements.in`, I we prevent them from being updated via the `pip-compile` mechanism.

Reply:
Ah, I misunderstood you. Of course, if the system depends on a fixed version than exact pinning is the right thing to do. But I think that's usually not the case, usually you want at least a certain version for some feature, and sometimes when a bug is introduced a version smaller than that bug.

Reply:
We had a couple of issues with pip dependencies that got upgraded and failed some ci jobs (last time pylint). For a stable setup I would prefer to actually have all dependencies pinned and a regular task to check for upgrades (but then on a branch without influencing the complete ci).

Reply:
In this example (I don't know why it was pinned, but let's assume it needs at least 0.29.0, but doesn't work with 0.31.x), we could write `>=0.29.0,<0.31.0`. Thus we would still receive patchlevel upgrades for `0.30` while still not updating to a version that doesn't work for us. And (to spread some FUD) I can imagine that some package maintainer delete old releases from PyPI eventually, so at some point `0.30.2` might not be available anymore (but `0.30.9` might still be available).

Reply:
Is it ok for you to move this out of scope and to https://github.com/magma/magma/issues/11743 -> "State of requirements.in should be validated vs setup.pys."?

Reply:
Yes, sounds good!